### PR TITLE
Notices improvements

### PR DESCRIPTION
**Pull request by paolochiodi**
This does two things as requested by @colmharte:
- convert `created_at` to timestamp
- add rate limiting for the `POST notices/create` endpoint 

The rate limiting has been completely lifted from the callback endpoints, but please @jackmurdoch double check the queries as I simplified one (also made the same change to the callback one)

### CODE REVIEW

File: lib/routes/callback/query.js [hunk 1]
```diff
@@ -41,9 +41,9 @@ const registerRateLimitedCallback = ({
                 ELSE callback_request_total + 1
                END
       WHERE   id = ${registrationId}
-        AND   (    last_callback IS NULL 
+        AND   (    last_callback IS NULL
                 OR last_callback <= CURRENT_TIMESTAMP - ${`${rateLimitSeconds} secs`}::INTERVAL
-                OR (last_callback > CURRENT_TIMESTAMP - ${`${rateLimitSeconds} secs`}::INTERVAL AND callback_rate_count < ${rateLimitRequestCount})
```
Comment (jackmurdoch) [inline lib/routes/callback/query.js:46]: Not sure what effect this will have - do you mean to change this line?

Comment (paolochiodi) [reply lib/routes/callback/query.js:46]: It should not have any effect, but leaving it in feels redundant. 

That is, if the `last_callback` is not null and not `<= CURRENT_TIMESTAMP` it surely is `> CURRENT_TIMESTAMP` so we can avoid a condition and only check `callback_rate_count`.

I guess it's mostly cosmetic.

Comment (colmharte) [reply lib/routes/callback/query.js:46]: Don't think we need rate limt time period and rate limit count. Rate limit time period should be sufficent

**Comment by paolochiodi** [comment]
@colmharte @jackmurdoch updated removing rate limit count for notices.

Now I have three modes:
- `noticesRateLimitSeconds` is null or not found: limit to one notice per registration
- `noticesRateLimitSeconds` is 0: no limit
- `noticesRateLimitSeconds` is any number: one request each `noticesRateLimitSeconds` seconds

**Review by jackmurdoch — APPROVED** [review]

**Comment by paolochiodi** [comment]
@colmharte @jackmurdoch updated with metrics

File: lib/routes/notices/index.js [hunk 2]
```diff
@@ -13,6 +15,63 @@ const {
...
+
+function logRateLimit(rateLimitSeconds) {
+  if (rateLimitSeconds == null) {
+    console.log(
+      'Notice Rate Limit: Off - users may only send a single notice key request'
+    )
+    return
+  }
+
+  if (rateLimitSeconds === 0) {
```
Comment (colmharte) [inline lib/routes/notices/index.js]: We don't want this option

File: lib/routes/notices/index.js [hunk 3]
```diff
@@ -13,6 +15,63 @@ const {
   free,
   isAvailable
 } = require('./query')
+const { metricsInsert } = require('../metrics/query')
+
+function logRateLimit(rateLimitSeconds) {
+  if (rateLimitSeconds == null) {
```
Comment (colmharte) [inline lib/routes/notices/index.js]: This should not be an option

File: lib/routes/notices/index.js [hunk 4]
```diff
@@ -13,6 +15,63 @@ const {
...
+function getCheckRateLimitFn(server, rateLimitSeconds) {
+  if (rateLimitSeconds == null) {
+    return rateLimitOnce.bind(null, server)
+  }
+
+  if (rateLimitSeconds === 0) {
+    return noop
+  }
+
+  return rateLimit.bind(null, server, rateLimitSeconds)
```
Comment (colmharte) [inline lib/routes/notices/index.js]: This is the only rate limit function we need

File: migrations/202011051700.do.notices-rate-limiter.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE registrations
+ADD COLUMN IF NOT EXISTS last_notice TIMESTAMPTZ NULL,
+ADD COLUMN IF NOT EXISTS notice_request_total INTEGER NULL;
```
Comment (colmharte) [inline migrations/202011051700.do.notices-rate-limiter.sql]: Not needed

**Review by colmharte — CHANGES_REQUESTED** [review]

**Comment by paolochiodi** [comment]
@colmharte updated with further simplification of rate limit and removed noticesToken

**Review by colmharte — APPROVED** [review]

**Pull request merged**